manifests/vhost/auth_cas.pp

```diff
@@ -0,0 +1,60 @@
+
+define apache::vhost::auth_cas (
+  $cas_attribute_prefix                                                             = undef,
+  $cas_attribute_delimiter                                                          = undef,
+  $cas_scrub_request_headers                                                        = undef,
+  $cas_sso_enabled                                                                  = undef,
+  $cas_login_url                                                                    = undef,
+  $cas_validate_url                                                                 = undef,
+  $cas_validate_saml                                                                = undef,
+  $cas_authoritative                                                                = undef,
+  $cas_cache_clean_interval                                                         = undef,
+  $cas_certificate_path                                                             = undef,
+  $cas_cookie_domain                                                                = undef,
+  $cas_cookie_entropy                                                               = undef,
+  $cas_cookie_http_only                                                             = undef,
+  $cas_cookie_path                                                                  = undef,
+  $cas_debug                                                                        = undef,
+  $cas_idle_timeout                                                                 = undef,
+  $cas_proxy_validate_url                                                           = undef,
+  $cas_root_proxied_as                                                              = undef,
+  $cas_timeout                                                                      = undef,
+  $cas_validate_depth                                                               = undef,
+  $cas_version                                                                      = undef,
+  $vhost                                                                            = $name,
+) {
+
+  # Is apache::mod::auth_cas enabled (or apache::mod['auth_cas'])
+  $cas_enabled = defined(Apache::Mod['auth_cas'])
```

Comment:
I'd argue that if you include this segment that you actually want to use cas so a ```include apache::mod::auth_cas``` would make sense to me.

Reply:
The answer is that I can't just `include apache::mod::auth_cas` because that class has required parameters, and I can't declare it myself because it has one parameter (`suppress_warnings`) that is not provided to the `vhost` defined type.